lib/src/message.dart

```diff
@@ -59,16 +58,30 @@ abstract class Message {
   /// Content-Type header, it will be set to "application/octet-stream".
   Message(body,
       {Encoding encoding,
-      Map<String, String> headers,
-      Map<String, Object> context})
-      : this._(new Body(body, encoding), headers, context);
+        Map<String, String> headers,
+        Map<String, Object> context})
+      : this.__(body, _determineMediaType(body, encoding, headers), headers,
+      context);
 
-  Message._(Body body, Map<String, String> headers, Map<String, Object> context)
+  Message.__(body, MediaType contentType, Map<String, String> headers,
```

Comment:
`__` isn't a good name for a constructor--it doesn't communicate what the difference is between this and the other unnamed constructor.

Reply:
K I think I need to rename this anyways so it could be used in the multipart implementation.

I also didn't like `fromValues` but didn't have any better ideas last night.